Metrics/start-windows-build.ps1

```diff
@@ -0,0 +1,365 @@
+    try {
+        if($CommitID) {
+            Start-ExternalCommand { git.exe reset --hard $CommitID } -ErrorMessage "Failed to set Metrics git repo last commit to: $CommitID."
+        }
+        Start-ExternalCommand { git.exe log -n 1 } -ErrorMessage "Failed to get the latest commit message for the Metrics git repo" | Out-File "$METRICS_BUILD_LOGS_DIR\latest-commit.log"
+        $commitId = Start-ExternalCommand { git.exe log --format="%H" -n 1 } -ErrorMessage "Failed to get the latest commit id for the dcos-metrics git repo"
+        Set-Variable -Name "LATEST_COMMIT_ID" -Value $commitId -Scope Global
+    } finally {
+        Pop-Location
+    }
+}
+
+function New-TestingEnvironment {
+    Write-Output "Creating new tests environment"
+    Start-EnvironmentCleanup # Do an environment cleanup to make sure everything is fresh
+    New-Directory $METRICS_DIR
+    New-Directory $METRICS_BUILD_OUT_DIR -RemoveExisting
+    New-Directory $METRICS_BUILD_LOGS_DIR
+    $global:PARAMETERS["BRANCH"] = $Branch
+    Start-GitClone -Path $METRICS_GIT_REPO_DIR -URL $GitURL -Branch $Branch
+    Set-LatestMetricsCommit
+    Start-GitClone -Path $METRICS_DCOS_WINDOWS_GIT_REPO_DIR -URL $DCOS_WINDOWS_GIT_URL
+    Start-ExternalCommand { git.exe config --global user.email "ostcauto@microsoft.com" } -ErrorMessage "Failed to set git user email"
+    Start-ExternalCommand { git.exe config --global user.name "ostcauto" } -ErrorMessage "Failed to set git user name"
+    Write-Output "New tests environment was successfully created"
+}
+
+function Start-DCOSMetricsBuild {
+    Write-Output "Building DC/OS Metrics in $METRICS_GIT_REPO_DIR"
+    Push-Location $METRICS_GIT_REPO_DIR
+    try {
+        New-Item -ItemType directory -Path ".\build" -Force
+        & $GOLANG_DIR\bin\go get .\...
+        Start-MetricsCIProcess  -ProcessPath "powershell.exe" `
+                                    -StdoutFileName "metrics-build-stdout.log" `
+                                    -StderrFileName "metrics-build-stderr.log" `
+                                    -ArgumentList @(".\scripts\build.ps1", "collector") `
+                                    -BuildErrorMessage "Metrics failed to build."
+        Copy-Item -Path "$METRICS_GIT_REPO_DIR\build\collector\dcos-metrics-collector-*" -Destination "$METRICS_GIT_REPO_DIR/dcos-metrics.exe"                                    
+    } finally {
+        Pop-Location
+    }
+    Write-Output "DC/OS Metrics was successfully built"
+}
+
+function New-DCOSMetricsPackage {
+    Write-Output "Creating DC/OS Metrics package"
+    Write-Output "METRICS_GIT_REPO_DIR: $METRICS_GIT_REPO_DIR"
+    New-Directory $METRICS_BUILD_BINARIES_DIR
+    $MetricsClusterIdFile = Join-Path $METRICS_BUILD_BINARIES_DIR "cluster-id"
+
+    Write-Output "Creating cluster-id file: $MetricsClusterIdFile"
+    # the following hardcoded cluster id was from the dcos-metrics source
+    $clusterid = "{fdb1d7c0-06cf-4d65-bb9b-a8920bb854ef}"
```

Comment:
Hm, what's up with this hard-coded cluster id ? A more verbose comment here will help as well.

Reply:
I will add more explanation on this.